backend/src/main/kotlin/org/loculus/backend/controller/SubmissionController.kt

```diff
@@ -254,36 +246,53 @@ open class SubmissionController(
                 description = "The total number of records sent in responseBody",
                 schema = Schema(type = "integer"),
             ),
+            Header(
+                name = "eTag",
+                description = "Last database write Etag",
+                schema = Schema(type = "integer"),
+            ),
         ],
     )
+    @ApiResponse(
+        responseCode = "304",
+        description = "No database changes since last request " +
+            "(Etag in HttpHeaders.IF_NONE_MATCH matches lastDatabaseWriteETag)",
+    )
     @GetMapping("/get-released-data", produces = [MediaType.APPLICATION_NDJSON_VALUE])
     fun getReleasedData(
         @PathVariable @Valid organism: Organism,
         @RequestParam compression: CompressionFormat?,
+        @Parameter(description = "(Optional) Only retrieve all released data if Etag has changed.")
+        @RequestHeader(value = HttpHeaders.IF_NONE_MATCH, required = false) ifNoneMatch: String?,
     ): ResponseEntity<StreamingResponseBody> {
-        val headers = HttpHeaders()
-        headers.contentType = MediaType.parseMediaType(MediaType.APPLICATION_NDJSON_VALUE)
-        if (compression != null) {
-            headers.add(HttpHeaders.CONTENT_ENCODING, compression.compressionName)
+        val lastDatabaseWriteETag = releasedDataModel.getLastDatabaseWriteETag(
+            RELEASED_DATA_RELATED_TABLES,
+        )
+        if (ifNoneMatch == lastDatabaseWriteETag) {
+            return ResponseEntity.status(HttpStatus.NOT_MODIFIED).build()
         }
 
+        val headers = HttpHeaders()
+        headers.eTag = lastDatabaseWriteETag
```

Comment:
Btw - this has the same issue as the `x-total-records`. In some edge cases it probably possible to:
* compute the etag
* change the data in the DB (changing the actual etag!)
* streaming the updated data with the old etag.

Reply:
Although not optimal I don't think this edge case is too bad as this means that we stream a more recent database state than would be expected by the etag. In the subsequent request the client will ask for new data using an etag which is older than the data it has. Potentially it will get the same data that it already obtained again but this is an edge case and again still much better than the current state. 

Of course if the order was reverted (i.e. stream data obtained before header calculated) this would be an enormous bug and would make this unusable.

Reply:
I also don't think that it is too bad.

Reply:
yes that's possible and not a problem - because it just causes one extra request in the future which is inconsequential.